File: TOOLS/lua/ontop-playback.lua
--makes mpv disable ontop when pausing and re-enable it again when resuming playback
--please note that this won't do anything if ontop was not enabled before pausing

local was_ontop = mp.get_property_native("ontop")

function f_load(event)
    if was_ontop then
        mp.set_property_native("ontop", true)
    end
end

mp.register_event("file-loaded", f_load)

mp.observe_property("pause", "bool", function(name, value)

local ontop = mp.get_property_native("ontop")
local is_idle = mp.get_property_native("playback-abort")
    if value then
        mp.set_property_native("ontop", false)
        was_ontop = ontop
    else
        if is_idle then
            mp.set_property_native("ontop", false)
            was_ontop = ontop
        else
            if was_ontop then
                mp.set_property_native("ontop", true)
            end
        end
    end
end)

File: TOOLS/python/ontop-playback.py
# makes mpv disable ontop when pausing and re-enable it again when resuming playback
# please note that this won't do anything if ontop was not enabled before pausing

from mpvclient import mpv  # type: ignore

was_ontop = False


@mpv.observe_property("pause", mpv.MPV_FORMAT_FLAG)
def func(value):
    ontop = mpv.get_property_node("ontop")
    global was_ontop
    if value:
        if ontop:
            mpv.set_property_node("ontop", False)
            was_ontop = True
    else:
        if was_ontop and not ontop:
            mpv.set_property_node("ontop", True)

        was_ontop = False
